Packet++/src/TcpLayer.cpp

```diff
@@ -366,8 +367,15 @@ namespace pcpp
 		const uint16_t portSrc = getSrcPort();
 		const char* payloadChar = reinterpret_cast<const char*>(payload);
 
-		if (HttpMessage::isHttpPort(portDst) &&
-		    HttpRequestFirstLine::parseMethod(payloadChar, payloadLen) != HttpRequestLayer::HttpMethodUnknown)
+		if ((DoIpLayer::isDoIpPort(portSrc) || DoIpLayer::isDoIpPort(portDst)) &&
+		    (DoIpLayer::isDataValid(payload, payloadLen)))
+		{
+			m_NextLayer = DoIpLayer::parseDoIpLayer(payload, payloadLen, this, m_Packet);
+			if (!m_NextLayer)
+				constructNextLayer<PayloadLayer>(payload, payloadLen, m_Packet);
+		}
+		else if (HttpMessage::isHttpPort(portDst) &&
```

Comment:
Since DoIP is more rare cpmaoring to HTTP, SSL, etc. - I suggest we don't set it as the first `if` but move it down to one of the `else if` - maybe somewhere around `SomeIpLayer`?